src/lib/api/prefetch.ts

```diff
@@ -0,0 +1,200 @@
+  whitelabel?: boolean;
+  transforms_ui?: boolean;
+};
+
+enum PrefetchOptionsEvents {
+  PICKER = 'picker',
+  TRANSFORM_UI = 'transform_ui',
+}
+
+type PrefetchOptions = {
+  pickerOptions?: PickerOptions;
+  settings?: PrefetchOptionsSetting;
+  permissions?: PrefetchOptionsPermissions;
+  events?: PrefetchOptionsEvents[];
+};
+
+interface SendObject {
+  apikey: string;
+  security?: {
+    policy?: string;
+    signature?: string;
+  };
+  permissions?: string[];
+  settings?: string[];
+  events?: PrefetchOptionsEvents[];
+  pickerOptions?: PickerOptions;
+}
+
+type responseObject = {
+  blocked?: boolean;
+  settings?: PrefetchOptionsSetting;
+  permissions?: PrefetchOptionsPermissions;
+  updated_config?: {
+    fromSources?: string[];
+  };
+};
+
+// ==>
+// {
+//   "apikey": "AHvhedybhQMqZOqRvZquez",
+//   security: {
+//     policy: "",
+//     signature: ""
+//   },
+// 	"permissions": ["transforms_ui", "gmail"], -- events
+// 	"settings": ["inapp_browser", "customsource"],
+// 	"events": ["picker"], -- events
+// 	"picker_config": {
+// 		"fromSources": ["googledrive", "dropbox"]
+// 	}
+// }
+
+// <==
+// {
+//   "blocked": false,
+//   "settings": {
+//     "customsource": false,
+//     "inapp_browser": true
+//   },
+//   "permissions": {
+//     "transforms_ui": false
+//   },
+//   "updated_config": {
+//     "fromSources": [
+//       "googledrive"
+//     ]
+//   }
+// }
+
+/**
+ * @private
+ */
+export class Prefetch {
+  private session: Session;
+
+  private prefetchUrl: string;
+
+  private configToCheck: PickerOptions;
+
+  constructor(session: Session) {
+    this.session = session;
+    this.prefetchUrl = session.urls.cloudApiUrl;
+  }
+
+  async getConfig({ pickerOptions, settings, permissions, events }: PrefetchOptions) {
+    if (this.session.prefetch) {
+      await requestWithSource().post(`${this.prefetchUrl}/prefetch`, {
+        events,
+      });
+
+      return Promise.resolve(this.session.prefetch);
+    }
+
+    const configToSend = this.cleanUpCallback(pickerOptions);
+
+    let paramsToSend: SendObject = {
+      apikey: this.session.apikey,
+      permissions: Object.keys(permissions),
+      settings: Object.keys(settings),
+      pickerOptions: configToSend,
+      events,
+    };
+
+    if (this.session.policy) {
+      Object.assign(paramsToSend, { security: { policy: this.session.policy } });
+    }
+
+    if (this.session.signature) {
+      Object.assign(paramsToSend, { security: { signature: this.session.signature } });
+    }
+
+    const response = await requestWithSource()
+      .post(`${this.prefetchUrl}/prefetch`, paramsToSend)
+      .then(res => res.data)
+      .then(data => data);
+
+    return this.reassignCallbacks(response);
+  }
+
+  private cleanUpCallback(pickerOptions: PickerOptions) {
+    this.configToCheck = cloneDeep.clone(pickerOptions);
+
+    const pickerDropPaneKey = ['onDragEnter', 'onDragLeave', 'onDragOver', 'onDrop', 'onSuccess', 'onError', 'onProgress', 'onClick'];
```

Comment:
move it to global costs the same with picker, you dont need to declare it every time function will be running or you cant remove it and check, if object key starts from ie: on -> so if (key.indexOf('on') === 0) then u dont need those constants

Reply:
done